package/kedro_viz/integrations/kedro/hooks_utils.py

```diff
@@ -0,0 +1,209 @@
+"""Utility functions for Kedro hooks implementation."""
+
+import json
+import logging
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+import fsspec
+from kedro.pipeline.node import Node as KedroNode
+
+from kedro_viz.constants import VIZ_METADATA_ARGS
+from kedro_viz.launchers.utils import _find_kedro_project
+from kedro_viz.utils import _hash, _hash_input_output
+
+logger = logging.getLogger(__name__)
+
+TIME_FORMAT = "%Y-%m-%dT%H.%M.%S.%fZ"
+EVENTS_DIR = VIZ_METADATA_ARGS["path"]
+EVENTS_FILE = "kedro_pipeline_events.json"
+
+
+def hash_node(node: Any) -> str:
+    """Stable ID for KedroNode or I/O reference."""
+    return _hash(str(node)) if isinstance(node, KedroNode) else _hash_input_output(node)
+
+
+def create_dataset_event(
+    event_type: str,
+    dataset_name: str,
+    dataset_value: Any = None,
+    datasets: Optional[Dict[str, Any]] = None,
+) -> Dict[str, Any]:
+    """Generic builder for dataset load/save events.
+
+    Args:
+        event_type: Event type/name
+        dataset_name: Dataset name
+        dataset_value: Dataset data
+        datasets: Dictionary of available datasets
+
+    Returns:
+        Dictionary with event data
+    """
+    event: Dict[str, Any] = {
+        "event": event_type,
+        "dataset": dataset_name,
+        "node_id": _hash_input_output(dataset_name),
+        "status": "Available",
```

Comment:
I see that we set the status of datasets later ? So I find it a bit confusing that we say 'Available' here and 'Missing' at another point in the api/responses